src/chat/ChatScreen.js

```diff
@@ -219,12 +221,18 @@ export default function ChatScreen(props: Props): Node {
               }
               showMessagePlaceholders={showMessagePlaceholders}
               startEditMessage={setEditMessage}
+              startQuoteAndReply={composeBoxImperativeMethods?.doQuoteAndReply ?? null}
             />
           );
         }
       })()}
       {showComposeBox && (
         <ComposeBox
+          // Why use a callback ref instead of useRef? So we can be notified
+          // when the ref payload changes. Why set state when it changes? To
+          // cause a rerender, giving MessageList, and so the action-sheet
+          // code, an up-to-date startQuoteAndReply prop from that state.
+          ref={setComposeBoxImperativeMethods}
```

Comment:
Huh, yeah I think that approach would work.

It fits well with the action sheet's current design where we don't live-update it when it's visible, e.g., to reflect a theme change or anything that would add/subtract buttons in the list.

If that weren't a given, I'd be wary of introducing a live-updating bug.

Reply:
True.

I think that design is something we can count on keeping. If the action sheet were to start having buttons sometimes pop into and out of existence, I think that would be annoying when it happens -- because it would move the other buttons around -- and would have no real benefit.

There's always the possibility that we offer some option that by the time the user chooses it isn't actually possible, or vice versa… but that possibility would still be there even if we live-updated the set of buttons. That's because the state that ultimately determines whether the action is possible is the state that's on the server — and moreover, the state that's on the server in the *future* when the request arrives there. So we might as well just freeze the set of options we offer to the user on the action sheet as the set that's based on the snapshot of state we have when they summon it.